File: .github/hooks/scripts/orchestrated_post_tool.py
from orchestrated_common import load_hook_input, load_state, print_json, repo_root_from_input, save_state, truncate_text, utc_now_iso


FILE_EDIT_TOOLS = {"apply_patch", "create_file", "edit_notebook_file"}


def main() -> int:
    input_data = load_hook_input()
    repo_root = repo_root_from_input(input_data)
    state = load_state(repo_root)
    tool_name = input_data.get("tool_name", "")
    tool_response = input_data.get("tool_response", "")

    state["toolAudit"].append(
        {
            "tool": tool_name,
            "timestamp": utc_now_iso(),
            "task": state.get("activeTaskName"),
        }
    )
    state["toolAudit"] = state["toolAudit"][-20:]

    additional_context = None
    system_message = None

    if tool_name in FILE_EDIT_TOOLS:
        additional_context = (
            f"orchestrated-agent audit: {tool_name} completed for task '{state.get('activeTaskName') or 'unspecified-task'}'."
        )

    if tool_name == "run_in_terminal":
        summary = truncate_text(str(tool_response), 180)
        additional_context = (
            f"orchestrated-agent audit: terminal execution completed for '{state.get('activeTaskName') or 'unspecified-task'}'. "
            f"Summary: {summary}"
        )

    if not state.get("feedbackHintShown") and state.get("activeTaskName"):
        state["feedbackHintShown"] = True
        system_message = (
            "orchestrated-agent: record feedback with 'rate great', 'rate okay', or 'rate not good'; "
            "use 'stop this request' to block further tool use."
        )

    save_state(repo_root, state)

    response = {}
    if system_message:
        response["systemMessage"] = system_message
    if additional_context:
        response["hookSpecificOutput"] = {
            "hookEventName": "PostToolUse",
            "additionalContext": additional_context,
        }
    return print_json(response)


if __name__ == "__main__":
    raise SystemExit(main())

File: .github/hooks/scripts/orchestrated_post_tool.mjs
import {
  loadHookInput,
  loadState,
  printJson,
  repoRootFromInput,
  runCli,
  saveState,
  truncateText,
  utcNowIso,
} from './orchestrated_common.mjs';

const COOLDOWN_TOOLS = new Set(['run_in_terminal', 'runSubagent', 'search_subagent']);
const FILE_EDIT_TOOLS = new Set(['apply_patch', 'create_file', 'edit_notebook_file']);

export async function main() {
  const inputData = await loadHookInput();
  const repoRoot = repoRootFromInput(inputData);
  const state = await loadState(repoRoot);
  const toolName = typeof inputData.tool_name === 'string' ? inputData.tool_name : '';
  const toolResponse = inputData.tool_response;

  state.toolAudit.push({
    tool: toolName,
    timestamp: utcNowIso(),
    task: state.activeTaskName,
  });
  state.toolAudit = state.toolAudit.slice(-20);

  if (COOLDOWN_TOOLS.has(toolName)) {
    state.lastTaskTimestamp = Date.now();
  }

  let additionalContext;
  let systemMessage;

  if (FILE_EDIT_TOOLS.has(toolName)) {
    additionalContext = `orchestrated-agent audit: ${toolName} completed for task '${state.activeTaskName ?? 'unspecified-task'}'.`;
  }

  if (toolName === 'run_in_terminal') {
    const summary = truncateText(String(toolResponse ?? ''), 180);
    additionalContext =
      `orchestrated-agent audit: terminal execution completed for '${state.activeTaskName ?? 'unspecified-task'}'. ` +
      `Summary: ${summary}`;
  }

  await saveState(repoRoot, state);

  const response = {};
  if (systemMessage) {
    response.systemMessage = systemMessage;
  }
  if (additionalContext) {
    response.hookSpecificOutput = {
      hookEventName: 'PostToolUse',
      additionalContext,
    };
  }
  return printJson(response);
}

await runCli(main);